device/peer_test.go

```diff
@@ -24,6 +25,19 @@ func checkAlignment(t *testing.T, name string, offset uintptr) {
 // hard segfault at runtime.
 func TestPeerAlignment(t *testing.T) {
 	var p Peer
+
+	typ := reflect.TypeOf(p)
+	t.Logf("Peer type size: %d, with fields:", typ.Size())
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		t.Logf("\t%30s\toffset=%3v\t(type size=%3d, align=%d)",
+			field.Name,
+			field.Offset,
+			field.Type.Size(),
+			field.Type.Align(),
+		)
```

Comment:
Cool idea using reflection for this. We could assert Size==Align, but I guess not everything is accessed atomically so maybe that'd be overkill.

Reply:
Not everything is aligned right now so doing that would require adding a tiny bit of padding. I'm not opposed to that, but I think I'll leave that for another commit another time.